src/coq_lib.mli

```diff
@@ -0,0 +1,27 @@
+(* This file is licensed under The MIT License *)
+(* (c) MINES ParisTech 2018-2019               *)
+(*     Written by: Emilio Jesús Gallego Arias  *)
+
+open! Stdune
+
+type t
+
+val name : t -> Lib_name.t
+
+val src_dir : t -> Path.t
+val obj_dir : t -> Path.t
+
+val package : t -> Package.Name.t option
+
+module DB : sig
+
+  type lib = t
+  type t
+
+  val create_from_coqlib_stanzas
+    : (Path.t * Dune_file.Coq.t) list
+    -> t
+
+  val resolve : t -> Loc.t * Lib_name.t -> lib Or_exn.t
+
+end
```

Comment:
You can add `with lib := t` and change `type lib = t` to `type lib`. This will relieve you from adding the not very useful type alias in the implementation.

Reply:
Done thanks.